cli/pcluster/config_sanity.py

```diff
@@ -532,16 +534,59 @@ def validate(self, resource_type, resource_value):  # noqa: C901 FIXME
             ]:
                 self.__fail(resource_type, "Region %s is not supported with batch scheduler" % self.region)
 
+            # Check spot bid percentage
+            if "SpotPrice" in resource_value:
+                spot_price = int(resource_value["SpotPrice"])
+                if spot_price > 100 or spot_price < 0:
+                    self.__fail(resource_type, "Spot bid percentage needs to be between 0 and 100")
+
+            # Check sanity on desired, min and max vcpus
+            if (
+                "DesiredSize" not in resource_value
+                or "MinSize" not in resource_value
+                or "DesiredSize" not in resource_value
+            ):
+                # this should never occur
+                self.__fail(resource_type, "Both min, desired and max vcpus parameters must be set")
+
+            min_size = int(resource_value["MinSize"])
+            desired_size = int(resource_value["DesiredSize"])
+            max_size = int(resource_value["MaxSize"])
+
+            if desired_size < min_size:
+                self.__fail(resource_type, "Desired vcpus must be greater than or equal to min vcpus")
+
+            if desired_size > max_size:
+                self.__fail(resource_type, "Desired vcpus must be fewer than or equal to max vcpus")
+
+            if max_size < min_size:
+                self.__fail(resource_type, "Max vcpus must be greater than or equal to min vcpus")
+
             # Check compute instance types
             if "ComputeInstanceType" in resource_value:
+                compute_instance_type = resource_value["ComputeInstanceType"]
                 try:
                     s3 = boto3.resource("s3", region_name=self.region)
                     bucket_name = "%s-aws-parallelcluster" % self.region
                     file_name = "instances/batch_instances.json"
                     try:
                         file_contents = s3.Object(bucket_name, file_name).get()["Body"].read().decode("utf-8")
                         supported_instances = json.loads(file_contents)
-                        for instance in resource_value["ComputeInstanceType"].split(","):
+
+                        if "," not in compute_instance_type and "." in compute_instance_type:
+                            # if the type is not a list, and contains dot (nor optimal, nor a family)
+                            # validate instance type against max_vcpus limit
+                            vcpus = get_vcpus_from_pricing_file(self.region, compute_instance_type)
+                            if max_size < vcpus:
```

Comment:
Shall we check if vcpus is -1 and print invalid instance type in that case? what if it's a new insstance type? we periodically update our list, right?

Reply:
fixed